File: template-api/Models/Task.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.Models
{
    public class Task
    {
        [Key]
        [Required]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public Priority Priority { get; set; }
        [Required]
        public Status Status { get; set; }
    }
}

File: react-frontend/src/components/Task.js
import React from "react";


const Task = ({ task, onDelete, onEdit, optionsPriority, optionsStatus }) => {
  return (
    <div className="task">
      <ul className="task">
        <li>Name: {task.name}</li>
        <li>Description: {task.description}</li>
        <li value={task.priority}> Priority: {optionsPriority.find((x)=>x.value == task.priority).label}</li>
        <li value={task.status}>Status: {optionsStatus.find((t)=>t.value == task.status).label}</li>
        {task.status !== 2 ? (
          <button className="btn" onClick={() => onEdit(task)}>
            EDIT
          </button>
        ) : (
          ""
        )}
        <button className="btn" onClick={() => onDelete(task.id)}>
          DELETE
        </button>
      </ul>
    </div>
  );
};

export default Task;
